src/main/kotlin/org/rust/ide/docs/RsDocumentationProvider.kt

```diff
@@ -72,22 +72,31 @@ class RsDocumentationProvider : AbstractDocumentationProvider() {
 private fun RsDocAndAttributeOwner.header(usePreTag: Boolean): String {
     val rawLines = when (this) {
         is RsFieldDecl -> listOfNotNull((parent?.parent as? RsDocAndAttributeOwner)?.presentableQualifiedName)
+        is RsStructOrEnumItemElement, is RsTraitItem -> listOfNotNull(presentableQualifiedModName)
         is RsFunction -> {
             val owner = owner
             when (owner) {
-                is RsFunctionOwner.Foreign, is RsFunctionOwner.Free -> listOfNotNull(presentableQualifiedModName)
-                is RsFunctionOwner.Impl ->
-                    listOfNotNull(presentableQualifiedModName) + owner.impl.declarationText
+                is RsFunctionOwner.Foreign,
+                is RsFunctionOwner.Free -> listOfNotNull(presentableQualifiedModName)
+                is RsFunctionOwner.Impl -> listOfNotNull(presentableQualifiedModName) + owner.impl.declarationText
                 is RsFunctionOwner.Trait -> owner.trait.declarationText
             }
         }
-        is RsStructOrEnumItemElement, is RsTraitItem -> listOfNotNull(presentableQualifiedModName)
+        is RsConstant -> {
+            val owner = owner
+            when (owner) {
+                is RsConstantOwner.Foreign,
```

Comment:
Hm, maybe we want to unity `Rs*Owner` classes to unify work with them? I'm not sure that it's very good decision because type alias owner can't be foreign

Reply:
I initially tried to do that while creating this series but abandoned the idea because i hit a lot of problems and code churn at that time. This time i tried again and it went much more smoothly.
And apart of the size of the code change i think its a good idea.